array.rb

```diff
@@ -1,4 +1,19 @@
 class Array
+  # call-seq:
+  #   add(item) -> item
+  #
+  # Adds an item to +self+ and returns the item itself.
+  # Returns the added item:
+  #
+  #   a =          [1, 2, 3]
+  #   a.add(4) #=> 4
+  #   a        #=> [1, 2, 3, 4]
+  #
+  # Related: see {Methods for Assigning}[rdoc-ref:Array@Methods+for+Assigning].
+  def add item
+    item.tap { push(it) }
```

Comment:
Why using unnecessary `tap`?

```suggestion
    self << item
    item
```

or, to avoid side effects by redefinition of other Array methods:

```suggestion
    Primitive.cexpr!(%q{rb_ary_push(self, item)})
    item
```